File: app/src/main/java/ir/programmerplus/realtime_example/Application.java
package ir.programmerplus.realtime_example;


import android.util.Log;

import java.util.concurrent.TimeUnit;

import androidx.multidex.MultiDexApplication;
import ir.programmerplus.realtime.RealTime;

public class Application extends MultiDexApplication {

    private static final String TAG = Application.class.getSimpleName();

    @Override
    public void onCreate() {
        super.onCreate();

        RealTime.builder(this)
                .withGpsProvider()
                .setLoggingEnabled(BuildConfig.DEBUG)
                .setSyncBackoffDelay(5, TimeUnit.MINUTES)
                .build(date -> Log.d(TAG, "RealTime is initialized, current dateTime: " + date));
    }
}

File: app/src/main/java/com/opensource/gpstime/Application.kt
package com.opensource.gpstime

import android.util.Log
import androidx.multidex.MultiDexApplication
import com.opensource.gpstime.realtime.RealTime
import java.util.concurrent.TimeUnit

class Application : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        RealTime.builder(this)
            .withGpsProvider()
            .setLoggingEnabled(BuildConfig.DEBUG)
            .setSyncBackoffDelay(5, TimeUnit.MINUTES)
            .build { date -> Log.d(TAG, "RealTime is initialized, current dateTime: $date") }
    }

    companion object {
        private const val TAG = "Application"
    }
}
